src/cljs/org/broadinstitute/firecloud_ui/page/billing/create_project.cljs

```diff
@@ -70,29 +63,31 @@
                     #(react/call :create-billing-project this))}])
    :component-did-mount
    (fn [{:keys [this]}]
-     (react/call :get-billing-accounts this)
-     ; register a function in the JS window
-     ; this window's child can access the function by calling its window.opener
-     (aset js/window sign-in/handler-fn-name
-           (fn [message]
-             (let [token (get message "access_token")]
-               (reset! utils/access-token token)
-               (utils/set-access-token-cookie token)
-               (react/call :get-billing-accounts this)))))
-   :component-will-unmount
-   (fn []
-     (js-delete js/window sign-in/handler-fn-name))
+     (react/call :get-billing-accounts this))
+   :.enable-billing-permissions
+   (fn [{:keys [this]}]
+     (-> @utils/google-auth2-instance
+         (.grantOfflineAccess (clj->js {:redirect_uri "postmessage"
+                                        :scope "https://www.googleapis.com/auth/cloud-billing"}))
+         (.then (fn [response]
+                  ;; At some point, the access token gets replaced with a new one, but it doesn't
+                  ;; happen right away, so this dummy delay is necessary.
+                  (js/setTimeout
+                   #(react/call :get-billing-accounts this)
+                   1000)))))
```

Comment:
1. According to [this documentation](https://developers.google.com/identity/protocols/OAuth2WebServer#incrementalAuth), adding the new scope does not invalidate the existing tokens. I don't believe it is generating a new token. It is giving me a code that could be used for that, but I don't see a reason to re-create the refresh token to include the new scope.
2. Google issues a request perhaps to update its token, but it appears that the `then` is fired before that request comes back. I might be able to save the previous value of the access token and loop until it has changed. I'll give that a try.